### PR TITLE
Minimum Steps to Collect Coins Added

### PR DESCRIPTION
**Pull request by MonalikaKapoor**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Minimum Steps To Collect Coin Stacks

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

### CODE REVIEW

File: C-Plus-Plus/cp/Minimum_steps_to_collect_coin_stacks.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    and returns the minimum number of steps
+    using height array
+*/
+int minimumSteps(int height[], int N) { 
+	return minimumStepsUtil(0, N, 0, height); 
+} 
+
+// Testing minimumSteps() method
+int main() { 
+	int height[] = { 2, 1, 2, 5, 1 };
```
Comment (siddharth25pandey) [inline C-Plus-Plus/cp/Minimum_steps_to_collect_coin_stacks.cpp]: Use Dynamic Input

Comment (siddharth25pandey) [inline C-Plus-Plus/cp/Minimum_steps_to_collect_coin_stacks.cpp]: Display a message in print form as what are you taking input from the user for better UI

File: C-Plus-Plus/cp/Minimum_steps_to_collect_coin_stacks.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+int minimumSteps(int height[], int N) { 
+	return minimumStepsUtil(0, N, 0, height); 
+} 
+
+// Testing minimumSteps() method
+int main() { 
+	int height[] = { 2, 1, 2, 5, 1 }; 
+	int N = sizeof(height) / sizeof(int); 
+
+	cout << minimumSteps(height, N) << endl;
```
Comment (siddharth25pandey) [inline C-Plus-Plus/cp/Minimum_steps_to_collect_coin_stacks.cpp]: Display a message in print form as what are you showing output to the user for better UI

File: C-Plus-Plus/cp/Minimum_steps_to_collect_coin_stacks.cpp [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+int main() { 
+	int height[] = { 2, 1, 2, 5, 1 }; 
+	int N = sizeof(height) / sizeof(int); 
+
+	cout << minimumSteps(height, N) << endl; 
+	return 0; 
+}
+
+/* Time Complexity: O(n^2)O(n
+​2
```
Comment (siddharth25pandey) [inline C-Plus-Plus/cp/Minimum_steps_to_collect_coin_stacks.cpp]: Also write Space Complexity

File: C-Plus-Plus/cp/Minimum_steps_to_collect_coin_stacks.cpp [hunk 4]
```diff
@@ -0,0 +1,72 @@
+/*Problem Statement: “Given an integer arrays representing the height of each stack of coins and the number of coin stacks, calculate the minimum number of straight lines that pass through all the coins (minimum number of steps to collect these coins).”*/
+
+
+// Hints: Divide and Conquer
+#include <iostream>
```
Comment (siddharth25pandey) [inline C-Plus-Plus/cp/Minimum_steps_to_collect_coin_stacks.cpp]: Use **bits/stdc++**

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@MonalikaKapoor Do the suggested changes and also add the update in the **README** file

File: C-Plus-Plus/cp/Minimum_steps_to_collect_coin_stacks.cpp (hunk 4 above)
Comment (MonalikaKapoor) [reply C-Plus-Plus/cp/Minimum_steps_to_collect_coin_stacks.cpp]: I've added all the changes as requested in PR #7507. Please review asap and merge if all's well!

**Review by ankitaggarwal23 — COMMENTED** [review]
Thank you for your valuable time.😊

**Comment by MonalikaKapoor** [comment]
Not to be merged? @ankit

**Comment by ankitaggarwal23** [comment]
Kindly not to open a new PR do the changes in that only.

**Comment by MonalikaKapoor** [comment]
I have made the changes in previous PR kindly check please

**Comment by MonalikaKapoor** [comment]
#7274 is updated with all the final changes for issue #7273

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Do the changes so that we can merge and finalized the score.

File: C-Plus-Plus/cp/Minimum_steps_to_collect_coin_stacks.cpp [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+ through all the coins (minimum number of steps to collect these coins).”*/
+
+// Hints: Divide and Conquer
+#include <bits/stdc++.h>
+using namespace std;
+
+// returns minimum of two numbers
+int min2(int a, int b) {
+  return a < b ? a : b;
+}
```
Comment (siddharth25pandey) [inline C-Plus-Plus/cp/Minimum_steps_to_collect_coin_stacks.cpp:12]: Min is already available in C++ in built Function
Directly use **min** & remove all this

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@MonalikaKapoor Small change is required, rest is fine

**Pull request closed without merging**